### PR TITLE
Add string filtering to team / team member filter dropdowns

### PR DESCRIPTION
**Pull request by jmtaber129**
fixes https://github.com/ParabolInc/parabol/issues/5777.

Adds search functionality to the team / team member filter dropdowns.  This is done by creating a couple of new things:
* A `SearchMenuItem` component - a styled menu label that accepts text input
* A `useSearchFilter` hook - A hook that takes in an array of items and a function for getting a string value of those items, and returns:
  * An `onQueryChange` handler - Takes a query string.  Pass this as `SearchMenuItem`'s `onChange` handler, or any other text input.
  * `filteredItems` - An array of items that match the query string
  * `query` - The last query passed to `onQueryChange`.
  
These patterns are based on the [filtering logic](https://github.com/ParabolInc/parabol/blob/2d134896d2da154b34fc80a5025022a63cd35a4b/packages/client/components/GitHubScopingSearchFilterMenu.tsx#L145-L154)+[components](https://github.com/ParabolInc/parabol/blob/2d134896d2da154b34fc80a5025022a63cd35a4b/packages/client/components/GitHubScopingSearchFilterMenu.tsx#L172-L177) for integrations mentioned in the original ticket.

This also refactors several existing search filters to use this component+hook:
* TaskFooterIntegrateMenuList.tsx
* GitHubScopingSearchFilterMenu.tsx
* JiraScopingSearchFilterMenu.tsx
* NewGitHubIssueMenu.tsx
* NewJiraIssueMenu.tsx

(Each of these refactors have been tested with local GH + Atlassian integrations)

Screen Shot 2022-01-26 at 2 40 51 PM

**Testing**
New search filtering:
* /me/tasks
  * Open the team dropdown (defaults to "All teams"), search for a team
  * Open the team member filter dropdown (defaults to current user), and search for a team member
  * Add a task to the board, click the task card's team, and search for a different team
* /team/:teamId
  * Open the team member filter dropdown ("all team members"), and search for a team member.  Confirm that team members are filtered and "All team members" is not shown when there's a search query.
  * Add a task card, click the task card's team member, and search for a different team member

Regression testing (for the refactors on existing search filtering):
* /me/tasks
  * Click the task card's "push to integration" button and search for GH and Atlassian integrations
* Sprint poker scoping
  * Filter by GH integration in scoping view
  * Filter by Jira integration in scoping view (search only shows up when more than 10 integrations are available)
  * For both Jira and GH, click new issue, and search for a different integration

### CODE REVIEW

File: packages/client/components/UserDashTeamMenu.tsx [hunk 1]
```diff
@@ -30,33 +32,47 @@ const UserDashTeamMenu = (props: Props) => {
...
+      defaultActiveIdx:
+        filteredTeams.findIndex((team) => teamIds?.includes(team.id)) + (showAllTeams ? 2 : 1)
     }
   }, [userIds, teamIds])
+
+  const {
+    query,
+    filteredItems: matchedFilteredTeams,
+    onQueryChange
+  } = useSearchFilter(filteredTeams!, (team) => team.name)
```
Comment (Dschoordsch) [inline packages/client/components/UserDashTeamMenu.tsx:49]: -1 please remove `!`

**Comment by jmtaber129** [comment]
Added a "No results" placeholder for when a search query is present, but there are no results.

Fixed those bugs - the `getValue` methods were returning non-lower-case strings, so it was matching a lower-case query to mixed case strings.  Fixed this by calling `toLowerCase()` on the result of `getValue` in the `useSearchFilter` hook

File: packages/client/components/UserDashTeamMemberMenu.tsx [hunk 2]
```diff
@@ -5,18 +5,31 @@ import Menu from './Menu'
...
+import styled from '@emotion/styled'
+import {PALETTE} from '~/styles/paletteV3'
+import MenuItemLabel from './MenuItemLabel'
 
 interface Props {
   menuProps: MenuProps
   viewer: UserDashTeamMemberMenu_viewer | null
 }
 
+const NoResults = styled(MenuItemLabel)({
```
Comment (Dschoordsch) [inline packages/client/components/UserDashTeamMemberMenu.tsx:25]: +1 You could put this component in a separate file as well and reuse it.

**Review by Dschoordsch — APPROVED** [review]
Works now. I have just one comment you don't need to address. I will move this forward to maintainer review. Good work!

**Comment by mattkrick** [comment]
This'll be my top PR to review tomorrow

**Comment by mattkrick** [comment]
Code looks super clean! Great abstraction using the hook & refactoring all the similar components.

Issue 1:
- Search box appears when there is only 1 person on the team. Maybe hide the search box if the total results are < 6. Business logic is up to you, share your thoughts. 
Screen Shot 2022-01-28 at 10 27 24 AM


Issue 2:
- The active value appears in the search results. Was this intended? With the existing logic, if "Matt" is the current assignee, then "Matt" is removed from the results (1 fewer thing to scroll). No strong opinion on this, but want to get your thoughts

Issue 3:
- Clicking a team member when scrollTop > 0 doesn't work (at least i _think_ that's what's going on?) repro here:

https://user-images.githubusercontent.com/5514175/151603971-be92c350-5668-4b27-b8b0-163b38b0ff24.mov

**Review by mattkrick — CHANGES_REQUESTED** [review]
-1 little usability bug with clicking in the mid or bottom of the scroll container. I agree with Georg about the NoResults refactor. Aside from that, looks _clean_!

**Comment by jmtaber129** [comment]
* Re: issue 1 - Changed to only show the search bar when there are more than 5 items, since this seems to be around when a scrollbox is necessary.
* Nice catch on issue 2 - This just needed to pass in the `keepParentFocus` flag to the `Menu` component to prevent the search box from keeping/stealing focus (if the search box was scrolled out of view, the search box would regain focus when any part of the menu is clicked, including when attempting to select an item).  Fixed.
* Refactored out the `NoResults` component into an `EmptyDropdownMenuItemLabel` component.

**Comment by mattkrick** [comment]
Retested & works great! 👍

**Pull request merged**